src/path-symex/path_symex.cpp

```diff
@@ -290,6 +297,103 @@ void path_symext::symex_malloc(
   assign(state, lhs, rhs);
 }
 
+
+/*******************************************************************\
+
+Function: get_old_va_symb
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+static irep_idt get_old_va_symb(
+    const path_symex_statet &state,
+    const exprt &src)
+{
+  if(src.id()==ID_typecast)
+    return get_old_va_symb(state, to_typecast_expr(src).op());
+  else if(src.id()==ID_address_of)
+  {
+    const exprt &op=to_address_of_expr(src).object();
+
+    if(op.id()==ID_symbol)
+    {
+      const irep_idt &identifier=to_symbol_expr(op).get_identifier();
+      return identifier;
+    }
+  }
+
+  return irep_idt();
+}
+
+
+/*******************************************************************\
+
+Function: path_symext::symex_va_arg_next
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void path_symext::symex_va_arg_next(
+    path_symex_statet &state,
+    const exprt &lhs,
+    const side_effect_exprt &code)
+{
+  if(code.operands().size()!=1)
+    throw "va_arg_next expected to have one operand";
+
+  if(lhs.is_nil())
+    return;// ignore
+
+  exprt tmp=state.read(code.op0()); // constant prop on va_arg parameter
+
+  // Get old symbol of va_arg and modify it to generate a new one.
+  irep_idt id=get_old_va_symb(state, tmp);
+  exprt rhs=gen_zero(lhs.type());
+
+  if(!id.empty())
+  {
+    // strip last name off id to get function name
+    std::size_t pos=id2string(id).rfind("::");
+    if(pos!=std::string::npos)
+    {
+      irep_idt function_identifier=std::string(id2string(id), 0, pos);
+      std::string base=id2string(function_identifier)+"::va_arg";
+
+      if(has_prefix(id2string(id), base))
+        id=base
+            +i2string(
+                safe_string2unsigned(
+                    std::string(id2string(id), base.size(), std::string::npos))
+                    +1);
+      else
+        id=base+"0";
```

Comment:
Ah, I see! Would you mind adding a comment to explain this? Long-term we might come up with a solution that uses appropriate data structures, but for now that's presumably the best we can do.